spec/wizards/placements/add_placement/steps/year_group_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe Placements::AddPlacement::Steps::YearGroup, type: :model do
+  describe "attributes" do
+    it { is_expected.to have_attributes(school: nil, year_group: nil) }
+  end
+
+  describe "delegations" do
+    it { is_expected.to delegate_method(:primary?).to(:school).with_prefix }
+  end
+
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:school) }
+
+    context "when the school is primary" do
+      subject { described_class.new(school:) }
+
+      let(:school) { instance_double(Placements::School) }
+
+      before { allow(school).to receive(:primary?).and_return(true) }
```

Comment:
Could we use a factory rather than a double here, please? 🙏🏻

I always feel uncomfortable about using doubles because they can end up subtly breaking things if the underlying object changes, and the test double doesn't get updated to match.

```suggestion
      let(:school) { build(:placements_school, phase: "Primary") }
```

It might also be a nice opportunity to add a `:primary` trait to the [schools factory](https://github.com/DFE-Digital/itt-mentor-services/blob/9102ee3579eec8dd7a2184e646a51d76f4f38518/spec/factories/schools.rb).